### PR TITLE
Hide inspector implementation details

### PR DESCRIPTION
**Pull request by eugeneo**
This should fix the AIX issues as STL headers will no longer leak outside.

This change was reviewed by +pavelfeldman

### CODE REVIEW

File: src/inspector_agent.cc [hunk 1]
```diff
@@ -78,7 +80,8 @@ void SendVersionResponse(inspector_socket_t* socket) {
       "[ {"
       "  \"Browser\": \"node.js/%s\","
       "  \"Protocol-Version\": \"1.1\","
-      "  \"User-Agent\": \"Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36"
+      "  \"User-AgentImpl\": \"Mozilla/5.0 (X11; Linux x86_64) "
```
Comment (ofrobots) [inline src/inspector_agent.cc]: nit: unnecessary style change?

File: src/inspector_agent.h [hunk 2]
```diff
@@ -5,27 +5,19 @@
...
-class V8Inspector;
-}
-
 // Forward declaration to break recursive dependency chain with src/env.h.
 namespace node {
 class Environment;
 }  // namespace node
 
+namespace v8 {
+  class Platform;
```
Comment (ofrobots) [inline src/inspector_agent.h]: nit: remove indent.

File: src/inspector_agent.cc (hunk 1 above)
Comment (ofrobots) [inline src/inspector_agent.cc]: on a closer look, did you intend to add 'Impl' suffix to 'User-Agent'?

Comment (eugeneo) [inline src/inspector_agent.cc]: Thanks for noticing! I did not realize where the lint error came from.

File: src/inspector_agent.h (hunk 2 above)
Comment (eugeneo) [inline src/inspector_agent.h]: Fixed

**Comment by eugeneo** [comment]
I uploaded another commit that addresses the comments. PTAL.

**Pull request closed without merging**